x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/RuleQueryBuilder.java

```diff
@@ -111,18 +111,6 @@ private RuleQueryBuilder(
             throw new IllegalArgumentException("rulesetId must not be null or empty");
         }
 
-        // PinnedQueryBuilder will return an error if we attempt to return more than the maximum number of
```

Comment:
@benwtrent had a great observation that these checks don't do anything because pinnedIds and docs are interstitial - moved these checks down into rewrite.

Reply:
They are so interstitial, they are never set in this constructor, as once we have them (or not) we return a different kind of query.

Could we remove them from the ctor here? Its private, so not that big of a deal, but we always pass `null`, so seems like we could remove them.

Reply:
I was going to remove them, but then ran into the tranpsort client changes, so figured I'd hold off for this PR...